exp/api/v1alpha3/types.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3"
+)
+
+// EBS can be used to automatically set up EBS volumes when an instance is launched.
+type EBS struct {
+	// Encrypted is whether the volume should be encrypted or not.
+	// +optional
+	Encrypted bool `json:"encrypted,omitempty"`
+
+	// The size of the volume, in GiB.
+	// This can be a number from 1-1,024 for standard, 4-16,384 for io1, 1-16,384
+	// for gp2, and 500-16,384 for st1 and sc1. If you specify a snapshot, the volume
+	// size must be equal to or larger than the snapshot size.
+	// +optional
+	VolumeSize int64 `json:"volumeSize,omitempty"`
+
+	// The volume type
+	// For more information, see Amazon EBS Volume Types (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html)
+	// +kubebuilder:validation:Enum=standard;io1;gp2;st1;sc1
```

Comment:
```suggestion
	// +kubebuilder:validation:Enum=standard;io1;io2;gp2;st1;sc1
```

Nit: io2 was added since the start of this PR. Can do as follow up

Reply:
Honestly I'm not even using EBS with our machinepools/ASG's. I think this struct fell here on day 1 while I was listing out "what might we need in the future". I can definitely delete it for now if that helps?

Reply:
Either add io2 now, or I can PR it next. Don't delete the validation though.